pkg/utils/eventrecorder/eventrecorder.go

```diff
@@ -16,84 +16,107 @@ package eventrecorder
 	awsNode      = "aws-node"
 	specNodeName = "spec.nodeName"
 	labelK8sapp  = "k8s-app"
+	EventReason  = sgpp.VpcCNIEventReason
 )
 
 type EventRecorder struct {
-	recorder  record.EventRecorder
-	k8sClient client.Client
+	Recorder        events.EventRecorder
+	RawK8SClient    client.Client
+	CachedK8SClient client.Client
 }
 
-func InitEventRecorder(k8sClient client.Client) error {
+func New(rawK8SClient, cachedK8SClient client.Client) (*EventRecorder, error) {
 
 	clientSet, err := k8sapi.GetKubeClientSet()
 	if err != nil {
 		log.Fatalf("Error Fetching Kubernetes Client: %s", err)
-		return err
+		return nil, err
 	}
-	eventBroadcaster := record.NewBroadcaster()
-	eventBroadcaster.StartRecordingToSink(&typedcorev1.EventSinkImpl{
-		Interface: clientSet.CoreV1().Events(""),
+	eventBroadcaster := events.NewBroadcaster(&events.EventSinkImpl{
+		Interface: clientSet.EventsV1(),
 	})
+	stopCh := make(chan struct{})
+	eventBroadcaster.StartRecordingToSink(stopCh)
 
-	recorder := &EventRecorder{}
-	recorder.recorder = eventBroadcaster.NewRecorder(clientgoscheme.Scheme, corev1.EventSource{
-		Component: awsNode,
-		Host:      myNodeName,
-	})
-	recorder.k8sClient = k8sClient
-	eventRecorder = recorder
-	return nil
-}
+	eventRecorder := &EventRecorder{}
+	eventRecorder.Recorder = eventBroadcaster.NewRecorder(clientgoscheme.Scheme, "aws-node")
+	eventRecorder.RawK8SClient = rawK8SClient
+	eventRecorder.CachedK8SClient = cachedK8SClient
+
+	return eventRecorder, nil
 
-func Get() *EventRecorder {
-	if eventRecorder == nil {
-		err := errors.New("error fetching event recoder, not initialized")
-		panic(err.Error())
-	}
-	return eventRecorder
 }
 
 // BroadcastEvent will raise event on aws-node with given type, reason, & message
 func (e *EventRecorder) BroadcastEvent(eventType, reason, message string) {
 
 	// Get aws-node pod objects with label & field selectors
 	labelSelector := labels.SelectorFromSet(labels.Set{labelK8sapp: awsNode})
-	fieldSelector := fields.SelectorFromSet(fields.Set{specNodeName: myNodeName})
+	fieldSelector := fields.SelectorFromSet(fields.Set{specNodeName: MyNodeName})
 	listOptions := client.ListOptions{
 		LabelSelector: labelSelector,
 		FieldSelector: fieldSelector,
 	}
 
 	var podList corev1.PodList
-	err := e.k8sClient.List(context.TODO(), &podList, &listOptions)
+	err := e.RawK8SClient.List(context.TODO(), &podList, &listOptions)
 	if err != nil {
 		log.Errorf("Failed to get pods, cannot broadcast events: %v", err)
 		return
 	}
 	for _, pod := range podList.Items {
 		log.Debugf("Broadcasting event on pod %s", pod.Name)
-		e.recorder.Event(&pod, eventType, reason, message)
+		e.Recorder.Eventf(&pod, nil, eventType, reason, "", message)
 	}
 }
+
+func (e *EventRecorder) findMyNode(ctx context.Context) (corev1.Node, error) {
+	var node corev1.Node
+	// Find my node
+	err := e.CachedK8SClient.Get(ctx, types.NamespacedName{Name: MyNodeName}, &node)
+	log.Debugf("Node found %s - labels - %d", node.Name, len(node.Labels))
```

Comment:
We're not checking for error here before we print this debug message. Will be misleading.

Reply:
Makes sense. I will add an error handle here.